conans/client/toolchain/meson.py

```diff
@@ -0,0 +1,297 @@
+import os
+import platform
+from configparser import ConfigParser
+from pathlib import Path
+
+from conans.client import tools
+from conans.errors import ConanException
+from conans.util.files import mkdir
+
+class MesonMachineFile():
+    """
+    A wrapper for meson machine files for the use in `MesonToolchain`.
+    Note: Meson requires that all ini values are quoted. E.g. `name = 'value'`
+    """
+    def __init__(self,
+                 name: str,
+                 path: str = None,
+                 config: ConfigParser = None):
+        if not name:
+            raise ConanException('`name` is empty: machine file must have a unique name supplied')
+        self.name = name
+
+        if path and config:
+            raise ConanException('Both `path` and `config` were supplied: only one should be used')
+        if path:
+            config = ConfigParser()
+            config.read(path)
+        self.options = config
+
+    def dump(self, install_dir: str):
+        outpath = Path(install_dir)
```

Comment:
Probably not worth supporting Python27 at this point, but ``Pathlib`` hasn't been introduced yet into the conan codebase, probably not introduce it for consistency.

Reply:
Sorry, I'm getting confused here: should I rewrite the code to use only Python 2.7? And if not and using Python 3.5+ is alright, then what would be the problem with `Pathlib` introduction? It's a (well-tested) part of standard library (and available since Python 3.4). I mean it's like using old Python string formatting syntax(`%s %s` and friends) when `'{}'.format(arg)` or `f'{arg}'` is available, i.e. it will make code only harder to read and more bug-prone.